src/v/redpanda/application.cc

```diff
@@ -1265,8 +1265,20 @@ application::set_proxy_client_config(ss::sstring name, std::any val) {
       });
 }
 
-void application::wire_up_and_start(::stop_signal& app_signal) {
+void application::wire_up_and_start(::stop_signal& app_signal, bool test_mode) {
```

Comment:
nit: as a test developer, I'd much rather have control over some activate_all_features flag, rather than a "test mode", otherwise, it's unclear in tests if I should ever set this to false. Yes it means that as we add more things to "test mode" we need to update more call sites, but it seems like that would be less accident-prone.

Also consider using a bool_class.